vllm/spec_decode/batch_expansion.py

```diff
@@ -247,24 +249,36 @@ def _create_target_seq_group_metadata(
         token_ids_to_score = self._get_token_ids_to_score(
             proposal_token_ids[batch_index])
 
+        # Use simpler sampling parameters apart from for final token
+        # (in particular don't do seeded sampling) since those sampled tokens
+        # aren't used
+        sampling_params = input_seq_group_metadata.sampling_params
+        non_bonus_sampling_params = DEFAULT_SIMPLE_SAMPLING_PARAMS \
+            if sampling_params.temperature else sampling_params
```

Comment:
Can you help me understand why we should use a default sampling params if the temperature is *not* None?

Reply:
I see, thx. I actually was digging into that part of the code the other day to understand how the rejection sampler worked in the greedy case, it makes sense. My confusion mainly stemmed from thinking temperature was optional.